mr/src/main/java/org/apache/iceberg/mr/Catalogs.java

```diff
@@ -77,6 +102,77 @@ private static Table loadTable(Configuration conf, String tableIdentifier, Strin
     return new HadoopTables(conf).load(tableLocation);
   }
 
+  /**
+   * Creates an Iceberg table using the catalog specified by the configuration.
+   * The properties should contain the following values:
+   * <p><ul>
+   * <li>Table identifier ({@link Catalogs#NAME}) or table path ({@link Catalogs#LOCATION}) is required
+   * <li>Table schema ({@link InputFormatConfig#TABLE_SCHEMA}) is required
+   * <li>Partition specification ({@link InputFormatConfig#PARTITION_SPEC}) is optional. Table will be unpartitioned if
+   *  not provided
+   * </ul><p>
+   * Other properties will be handled over to the Table creation. The controlling properties above will not be
+   * propagated.
+   * @param conf a Hadoop conf
+   * @param props the controlling properties
+   * @return the created Iceberg table
+   */
+  public static Table createTable(Configuration conf, Properties props) {
+    String schemaString = props.getProperty(InputFormatConfig.TABLE_SCHEMA);
+    Preconditions.checkNotNull(schemaString, "Table schema not set");
+    Schema schema = SchemaParser.fromJson(props.getProperty(InputFormatConfig.TABLE_SCHEMA));
```

Comment:
It looks like this is the reason why the examples specify a table property. Can we instead use Hive schema DDL and convert it to Iceberg? Similarly, can we get the identity partition fields that way to create a spec?

Reply:
I think we should keep the serialized schema for the Catalogs interface. Other systems like Impala, Presto, etc. might want to use it as well.
I would like to tackle the Hive schema DDL in another PR. The data is available in HiveIcebergSerDe.initialize in a somewhat convoluted way. I would like to get it there and convert it to the Iceberg Schema string. From there I would only push the Iceberg related stuff down further.
What do you think?

Reply:
I think it's fine to do this in a separate PR. I just really don't want to require setting properties with JSON schema or spec representations as the way to use Iceberg. It's okay for a way to customize if there isn't syntax, but normal cases should just use DDL.